modin/backends/pandas/query_compiler.py

```diff
@@ -2740,6 +2781,8 @@ def groupby_agg_builder(df, by=None, drop=False, partition_idx=None):
             def compute_groupby(df, drop=False, partition_idx=0):
                 """Compute groupby aggregation for a single partition."""
                 grouped_df = df.groupby(by=by, axis=axis, **groupby_kwargs)
+                if partition_selection is not None:
+                    grouped_df = grouped_df[partition_selection]
```

Comment:
Why move the `mask` logic here?

Reply:
it's not masking, it's actual __selection__. `grouped_df` here is a GroupBy object